File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int min = prices[0];

        int maxProfit = 0;

        for (int i = 1; i < prices.size(); i++) {
            int profit = 0;
            if (prices[i] <= min) {
                min = prices[i];
            }

            else {
                profit = prices[i] - min;
            }


            maxProfit = max(maxProfit, profit);
            
        }

        return maxProfit;
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        output = 0

        minPrice = prices[0]

        for i in range(1, len(prices)):
            profit = 0

            if prices[i] < minPrice:
                minPrice = prices[i]

            else:
                profit = prices[i] - minPrice
            
            output = max(output, profit)
        
        return output

        